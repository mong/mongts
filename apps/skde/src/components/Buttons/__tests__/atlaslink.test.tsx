/**
 * To view a rendered version of all the snapshots:
 *
 * Copy the contents of ./__snapshots__/index.tsx.snap
 * Go to https://codesandbox.io/s/snapshot-viewer-lqjjsm?file=/src/snapshots.js
 * Replace the contents with what you copied
 */

import { render } from "@testing-library/react";
import { AtlasLink } from "..";
import { vi } from "vitest";

vi.mock("next/font/google", () => ({
  Plus_Jakarta_Sans: () => ({
    weight: ["200", "300", "400", "500", "600", "700", "800"],
    subsets: ["latin"],
    display: "swap",
    style: {
      fontFamily: "Plus Jakarta Sans",
    },
  }),
}));

test("Norsk render", async () => {
  const date1 = new Date("October 13, 2014 11:13:00");
  const { container } = render(
    <AtlasLink
      lang="no"
      linkTo="v2/test_atlas"
      imageSource="https://picsum.photos/id/237/100/100.jpg"
      linkTitle="Testatlas 1"
      wide={false}
      linkText="Lorem lipsum dolor 1"
      date={date1}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("English render", async () => {
  const date1 = new Date("October 13, 2014 11:13:00");
  const { container } = render(
    <AtlasLink
      lang="en"
      linkTo="v2/en/test_atlas"
      imageSource="https://picsum.photos/id/237/100/100.jpg"
      linkTitle="Testatlas 2"
      wide={false}
      linkText="Lorem lipsum dolor 2"
      date={date1}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("Wide is true", async () => {
  const date1 = new Date("October 13, 2014 11:13:00");
  const { container } = render(
    <AtlasLink
      lang="no"
      linkTo=""
      imageSource="https://picsum.photos/id/237/100/100.jpg"
      linkTitle=""
      wide={true}
      linkText=""
      date={date1}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("newlyUpdated is true", async () => {
  const date1 = new Date("October 13, 2014 11:13:00");
  vi.useFakeTimers().setSystemTime(new Date("2014-10-10"));
  const { container } = render(
    <AtlasLink
      lang="no"
      linkTo=""
      imageSource="https://picsum.photos/id/237/100/100.jpg"
      linkTitle=""
      wide={false}
      linkText=""
      date={date1}
    />,
  );
  expect(container).toMatchSnapshot();
});
