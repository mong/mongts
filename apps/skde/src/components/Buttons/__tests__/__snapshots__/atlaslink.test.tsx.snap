// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`English render 1`] = `
<div>
  <div
    class="_linkOuterWrapper_6c6071 "
  >
    <a
      data-testid="v2/en/test_atlas"
      href="/v2/en/test_atlas"
    >
      <div
        class="_linkInnerWrapper_6c6071"
      >
        <div
          class="_linkImageWrapper_6c6071"
        >
          <img
            alt="atlas photo"
            data-nimg="1"
            decoding="async"
            height="407"
            loading="lazy"
            src="https://picsum.photos/id/237/100/100.jpg?w=1920&q=75"
            srcset="https://picsum.photos/id/237/100/100.jpg?w=640&q=75 1x, https://picsum.photos/id/237/100/100.jpg?w=1920&q=75 2x"
            style="color: transparent; max-width: 100%; height: auto; object-fit: cover;"
            width="610"
          />
        </div>
        <div
          class="_linkText_6c6071"
        >
          <div
            class="_linkDateContainer_6c6071 "
          >
            <div
              class="_outerCircle_6c6071"
            >
              <div
                class="_innerDot_6c6071"
              />
            </div>
            Published 
            <strong
              class="_date_6c6071"
            >
              October 13 2014
            </strong>
          </div>
          <div
            class="_linkTitle_6c6071"
          >
            Testatlas 2
          </div>
          <div
            class="_linkIngress_6c6071"
          >
            <p>
              Lorem lipsum dolor 2
            </p>
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`Norsk render 1`] = `
<div>
  <div
    class="_linkOuterWrapper_6c6071 "
  >
    <a
      data-testid="v2/test_atlas"
      href="/v2/test_atlas"
    >
      <div
        class="_linkInnerWrapper_6c6071"
      >
        <div
          class="_linkImageWrapper_6c6071"
        >
          <img
            alt="atlas photo"
            data-nimg="1"
            decoding="async"
            height="407"
            loading="lazy"
            src="https://picsum.photos/id/237/100/100.jpg?w=1920&q=75"
            srcset="https://picsum.photos/id/237/100/100.jpg?w=640&q=75 1x, https://picsum.photos/id/237/100/100.jpg?w=1920&q=75 2x"
            style="color: transparent; max-width: 100%; height: auto; object-fit: cover;"
            width="610"
          />
        </div>
        <div
          class="_linkText_6c6071"
        >
          <div
            class="_linkDateContainer_6c6071 "
          >
            <div
              class="_outerCircle_6c6071"
            >
              <div
                class="_innerDot_6c6071"
              />
            </div>
            Publisert:
            <strong
              class="_date_6c6071"
            >
              13.10.2014
            </strong>
          </div>
          <div
            class="_linkTitle_6c6071"
          >
            Testatlas 1
          </div>
          <div
            class="_linkIngress_6c6071"
          >
            <p>
              Lorem lipsum dolor 1
            </p>
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`Wide is true 1`] = `
<div>
  <div
    class="_linkOuterWrapper_6c6071 _wide_6c6071"
  >
    <a
      data-testid=""
      href="/"
    >
      <div
        class="_linkInnerWrapper_6c6071"
      >
        <div
          class="_linkImageWrapper_6c6071"
        >
          <img
            alt="atlas photo"
            data-nimg="1"
            decoding="async"
            height="407"
            loading="lazy"
            src="https://picsum.photos/id/237/100/100.jpg?w=1920&q=75"
            srcset="https://picsum.photos/id/237/100/100.jpg?w=640&q=75 1x, https://picsum.photos/id/237/100/100.jpg?w=1920&q=75 2x"
            style="color: transparent; max-width: 100%; height: auto; object-fit: cover;"
            width="610"
          />
        </div>
        <div
          class="_linkText_6c6071"
        >
          <div
            class="_linkDateContainer_6c6071 "
          >
            <div
              class="_outerCircle_6c6071"
            >
              <div
                class="_innerDot_6c6071"
              />
            </div>
            Publisert:
            <strong
              class="_date_6c6071"
            >
              13.10.2014
            </strong>
          </div>
          <div
            class="_linkTitle_6c6071"
          />
          <div
            class="_linkIngress_6c6071"
          >
            <p />
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`newlyUpdated is true 1`] = `
<div>
  <div
    class="_linkOuterWrapper_6c6071 "
  >
    <a
      data-testid=""
      href="/"
    >
      <div
        class="_linkInnerWrapper_6c6071"
      >
        <div
          class="_linkImageWrapper_6c6071"
        >
          <img
            alt="atlas photo"
            data-nimg="1"
            decoding="async"
            height="407"
            loading="lazy"
            src="https://picsum.photos/id/237/100/100.jpg?w=1920&q=75"
            srcset="https://picsum.photos/id/237/100/100.jpg?w=640&q=75 1x, https://picsum.photos/id/237/100/100.jpg?w=1920&q=75 2x"
            style="color: transparent; max-width: 100%; height: auto; object-fit: cover;"
            width="610"
          />
        </div>
        <div
          class="_linkText_6c6071"
        >
          <div
            class="_linkDateContainer_6c6071 _newlyUpdated_6c6071"
          >
            <div
              class="_outerCircle_6c6071"
            >
              <div
                class="_innerDot_6c6071"
              />
            </div>
            Publisert:
            <strong
              class="_date_6c6071"
            >
              13.10.2014
            </strong>
          </div>
          <div
            class="_linkTitle_6c6071"
          />
          <div
            class="_linkIngress_6c6071"
          >
            <p />
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
